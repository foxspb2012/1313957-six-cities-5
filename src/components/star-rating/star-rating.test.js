import React from 'react';
import renderer from 'react-test-renderer';
import StarRating from './star-rating';

const RAITING_TYPE = `place-card__stars`;

describe(`Should StarRating render correctly`, () => {
  it(`Should StarRating render correctly without typeClass and value 0`, () => {
    const tree = renderer
      .create(
          <StarRating
            value={0}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should StarRating render correctly without typeClass and value 2.5`, () => {
    const tree = renderer
      .create(
          <StarRating
            value={2.5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should StarRating render correctly without typeClass and value 5`, () => {
    const tree = renderer
      .create(
          <StarRating
            value={5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should StarRating render correctly with typeClass and value 0`, () => {
    const tree = renderer
      .create(
          <StarRating
            typeClass={RAITING_TYPE}
            value={0}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should StarRating render correctly with typeClass and value 2.5`, () => {
    const tree = renderer
      .create(
          <StarRating
            typeClass={RAITING_TYPE}
            value={2.5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should StarRating render correctly with typeClass and value 5`, () => {
    const tree = renderer
      .create(
          <StarRating
            typeClass={RAITING_TYPE}
            value={5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
