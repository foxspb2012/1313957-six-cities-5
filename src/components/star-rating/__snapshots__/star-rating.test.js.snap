// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should StarRating render correctly Should StarRating render correctly with typeClass and value 0 1`] = `
<div
  className="rating__stars place-card__stars"
>
  <span
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should StarRating render correctly Should StarRating render correctly with typeClass and value 2.5 1`] = `
<div
  className="rating__stars place-card__stars"
>
  <span
    style={
      Object {
        "width": "60%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should StarRating render correctly Should StarRating render correctly with typeClass and value 5 1`] = `
<div
  className="rating__stars place-card__stars"
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should StarRating render correctly Should StarRating render correctly without typeClass and value 0 1`] = `
<div
  className="rating__stars undefined"
>
  <span
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should StarRating render correctly Should StarRating render correctly without typeClass and value 2.5 1`] = `
<div
  className="rating__stars undefined"
>
  <span
    style={
      Object {
        "width": "60%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should StarRating render correctly Should StarRating render correctly without typeClass and value 5 1`] = `
<div
  className="rating__stars undefined"
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;
